templates/tapestry.js

```diff
@@ -901,6 +837,78 @@ function tapestryTool(config){
             rebuildNodeContents();
         }
     }
+
+    function initializeSelection() {
+        new DragSelect({
+            selectables: document.querySelectorAll(".node"),
+            onDragStart: () => {
+                if (!isMultiSelect) {
+                    selection.clear()
+                }
+            },
+            onElementSelect: node => {
+                const id = node.id.split("node-")[1]
+                selection.add(tapestry.dataset.nodes[findNodeIndex(id)])
+            },
+            onElementUnselect: node => {
+                const id = node.id.split("node-")[1]
+                selection.delete(tapestry.dataset.nodes[findNodeIndex(id)])
+            }
+        });
+    }
+
+    const selection = createSelection();
+    let isMultiSelect = false;
+
+    document.addEventListener("keydown", evt => {
+        if (evt.code === "Escape") {
+            selection.clear();
+        }
+        if (evt.ctrlKey || evt.shiftKey || evt.metaKey) {
+            isMultiSelect = true;
+            if (evt.code === "KeyA") {
+                evt.preventDefault();
+                tapestry.dataset.nodes.forEach(d => selection.add(d));
+            }
+        }
+    });
+
+    document.addEventListener("keyup", () => {
+        isMultiSelect = false;
+    });
+
+    function createSelection() {
+        const data = new Set();
+        const selection = {
+            size() {
+                return data.size;
+            },
+            add(node) {
+                data.add(node);
+                data.forEach(d => {
+                    const nd = document.getElementById(`node-${d.id}`);
+                    nd.classList.add("node-selected");
+                })
+            },
+            has(node) {
+                return data.has(node);
+            },
+            delete(node) {
+                data.delete(node);
+                const nd = document.getElementById(`node-${node.id}`);
+                nd.classList.remove("node-selected");
+            },
+            clear() {
+                data.forEach(d => {
+                    selection.delete(d);
+                })
+            },
+            forEach(fn) {
+                data.forEach(fn);
+            }
+        }
+        return selection;
+    }
```

Comment:
Can you please move this into where it says "ADD EDITOR ELEMENTS" and put the functions declarations all after the event listener additions?